src/scripts/choices.js

```diff
@@ -1787,7 +1779,7 @@ class Choices {
       userTemplates = callbackOnCreateTemplates.call(this, strToEl);
     }
 
-    this.config.templates = merge(TEMPLATES, userTemplates);
+    this.templates = merge(TEMPLATES, userTemplates);
```

Comment:
I believe `templates` should be private member (as consumer should not have direct access to it)? So, something like `_templates` by your notation...